pallets/file-system/src/lib.rs

```diff
@@ -313,11 +400,12 @@ pub mod pallet {
             let start_block = NextStartingBlockToCleanUp::<T>::get();
             let mut block_to_clean = start_block;
 
-            // Total weight used to avoid exceeding the remaining weight
+            // Total weight used to avoid exceeding the remaining weight.
+            // We count one write for the `NextBlockToCleanup` storage item updated at the end.
             let mut total_used_weight = Weight::zero();
 
             let required_weight_for_iteration =
-                db_weight.reads_writes(1, T::MaxExpiredStorageRequests::get().into());
+                db_weight.reads_writes(1, (T::MaxExpiredStorageRequests::get() + 1).into());
```

Comment:
```suggestion
                db_weight.reads_writes(1, (T::MaxExpiredStorageRequests::get() as u64 + 1u64).into());
```
```

Reply:
Since `MaxExpiredStorageRequests` is `Get<u32>`, this way you make sure that you will not overflow by casting and making the sum in `u64`. Anyway, actually the better way to do it would be a `checked_add` with an `expect_or_err!`, because if you later change `MaxExpiredStorageRequests` to be `Get<u64>` this trick won't save you anymore.